### DISCUSSION TITLE
Customize keymap

### DISCUSSION THREAD

**Discussion by frogfrogfrogfrogfrog** [monkeytypegame/monkeytype#862] [Ideas] [answered]
The keymap customization is kind of stale right now. You can't change the keymap in really any way besides how it reacts to your pressing and what keyboard layout it shows. So I've come up with a few ideas and why they might be helpful.

- Change keymap size because it's a bit small right now (at least in my opinion) 
- Limit what rows of the keymap you can see and what you can't and this could be a good feature for people who are currently learning keyboard layouts and want to look at a reference keymap but don't want to look for every single key
- Be able to customize how react works (delay time, how it reacts to a key being held down for long periods, etc)
- Another feature for react keymap would be to have a different color pop up if you pressed down the wrong key
- Be able to show a fuller keyboard (the 104 keys and maybe the 74 keys on the laptop keyboard)
- Maybe be able to customize the keymap itself with what color the keymap is or maybe it's opacity?

**Comment by Miodec** [comment, marked as answer]
It seems to me like a lot of overcomplicating for a very minor feature.

**Reply by frogfrogfrogfrogfrog** [reply]
Well here's my reason for showing you many ideas. I thought you could implement the ones you thought was reasonable and leave out the ones you didn't think was reasonable for the keymap. I know it's overcomplicating them if you were to add in every single idea so I thought it'd be reasonable for maybe one or two of them be implemented.

If I had to choose one that seemed reasonable it would be able to change the keymap size since I think it's too small.

If I wanted to add another mechanic to the keymap it would be limiting what you could see.

**Reply by frogfrogfrogfrogfrog** [reply]
Yeah I agree about that now.
